### DISCUSSION TITLE
device Müller Licht Remote 404022 - how to complete the integration

### DISCUSSION THREAD

**Discussion by ernstkl** [Koenkk/zigbee2mqtt#11913] [Devices]
Hello,

the device is listed as supported and I can successfully pair it with zigbee2mqtt.

But the support seems to lack some essential parts. I would be willing to put some work into this but I have no idea where to start. Maybe even the way the device works can not matched at all with how zigbee-herdsman handles things (groups).

I looked at the [code](https://github.com/Koenkk/zigbee-herdsman-converters/blob/fac271c69ecc722eee2f116f73c72dd9ed675a6c/devices/m%C3%BCller_licht.js#L150) -- it looks rather minimal to me.

So what is the actual "issue"?

When used in a standalone setup with bulbs of the same manufacturer the remote supports sending commands to three (or four) different groups. The target group is selected by repeatedly pressing a selection button before using the usual up/down buttons.

When paired to zigbee2mqtt and buttons are pressed I see action commands that have an extra attribute "action_group" which contains a value of 16387, 16388, ... depending on which group has been selected on the remote.

What I now would "expect" is that I can use zigbee2mqtt to bind these three (or four) "faces" of the tint remote to different groups I have defined in zigbee2mqtt. But the "faces" do not seem to appear in zigbee2mqtt in any way (I guess they should appear as endpoints of the device).

What _does_ work though is the following: when creating my bulb groups I use 16387 etc as group id's. This magically makes the bulb groups react on the corresponding "faces" of the remote.

Finally a remark about me writing "three or four" above. In the standalone setup the remote supports sending commands to either one selected group out of three possible groups or to all groups together. In zigbee2mqtt it looks like this is realized as four groups, because I see four different action_group values.

**Comment by heresiarch** [comment]
Hello I could confirm that behavior with the latest remote 404049d
https://www.zigbee2mqtt.io/devices/404049D.html#m%25C3%25BCller%2520licht-404049d
For that remote I had to define the Group ID of my Bulbs with the IDs from the action_group.
The older Müller Licht Remote does not that behavior and could be added to groups without Problems:
https://www.zigbee2mqtt.io/devices/MLI-404011_MLI-404049.html#m%25C3%25BCller%2520licht-mli-404011%252Fmli-404049

**Comment by heresiarch** [comment]
This is a workaround if you have many remotes of the same model you will have an overlap of the action groups.

**Comment by msaphoto06** [comment]
Hi,
Same thing with the EGLO 99099 remote control which sends to 3 groups.
I did like you, used the group ID, but that prevents me from buying a second remote control
